optimizely/optimizely.py

```diff
@@ -434,10 +432,8 @@ def _get_all_feature_variables_for_type(
         feature_enabled = False
         source_info = {}
 
-        user_context = self.create_user_context(user_id, attributes)
-        # error is logged in create_user_context
-        if user_context is None:
-            return None
+        user_context = OptimizelyUserContext(self, self.logger, user_id, attributes, False)
```

Comment:
We can add a unit test validating we do not fire ODP events with any legacy APIs.